File: DeepSpace/lib/weapon_type.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Deepspace
  module WeaponType
    class Type
      def initialize(power)
        @power=power
      end

      def power
        @power  
      end
    end

    LASER=Type.new(2.0)
    MISSILE=Type.new(3.0)
    PLASMA=Type.new(4.0)
  end
end

File: deepspace/src/deepspace/WeaponType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author antonio
 */
public enum WeaponType {
    
    LASER(2.0f),
    MISSILE(3.0f),
    PLASMA(4.0f);
    
    private float power;
    
    WeaponType(float power){
        this.power=power;
    }

    float getPower(){
        return power;
    }

}
